### PR TITLE
Analyse function feature

### PR DESCRIPTION
**Pull request by Korostast**
New bot command: `/analysis`. It have different functions that will be processed with Sympy. You can see patterns of analysis requests in file `/resources/analysis_patterns.md`.

- Now we have abstract class 'Parser' and two children: GraphParser and CalculusParser
- Add next math functions:
  - Derivative
  - Domain
  - Range
  - Zeros
  - Axis intersection
  - Periodicity
  - Convexity
  - Concavity
  - Continuity
  - Evenness / oddness
  - Vertical asymptotes
  - Horizontal asymptotes
  - Slant asymptotes
  - Stationary points
  - Maximum value
  - Minimum value
- CalculusParser can process queries with functions above
- Some graph parser fixes
- Updated .pylintrc
- New requirement: Python 3.10

### CODE REVIEW

File: source/core/handling_msg.py [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 from telegram import Update
 from telegram.ext import CallbackContext
 
+from source.conf import Config
+from source.math.calculus_parser import CalculusParser
 from source.math.graph import Graph
-from source.math.parser import Parser, ParseError
+from source.math.graph_parser import GraphParser, ParseError
+
+SETTINGS = Config()
```
Comment (RaEzhov) [inline source/core/handling_msg.py:17]:
```suggestion
PPI = '600'
```

File: source/core/handling_msg.py [hunk 2]
```diff
@@ -36,11 +42,48 @@ def send_graph(update: Update, context: CallbackContext):
...
+            update.message.reply_text("Couldn't find a suitable template. Check the input.")
+            return
+        result = parser.process_query()
+
+        # If USE_LATEX set in True, then send picture to the user. Else, send basic text
+        if SETTINGS.properties["APP"]["USE_LATEX"]:
+            latex = parser.make_latex(result)
+            with BytesIO() as latex_picture:
+                sy.preview(fr'${latex}$', output='png', viewer='BytesIO', outputbuffer=latex_picture,
+                           dvioptions=['-D', '600'])
```
Comment (RaEzhov) [inline source/core/handling_msg.py:75]:
```suggestion
                           dvioptions=['-D', PPI])
```

**Review by RaEzhov — APPROVED** [review]

File: source/core/handling_msg.py [hunk 3]
```diff
@@ -24,7 +30,7 @@ def send_graph(update: Update, context: CallbackContext):
         expr = " ".join(context.args)
```
[the remark this answers is not in the document]
Comment (Korostast) [reply source/core/handling_msg.py:30]: Right!

File: source/core/handling_msg.py (hunk 1 above)
Comment (Korostast) [reply source/core/handling_msg.py:17]: I'll do it later, ok

**Pull request merged**